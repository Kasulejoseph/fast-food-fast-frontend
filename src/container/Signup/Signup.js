import React from 'react';
import { connect } from 'react-redux';
import { registerAction} from '../../actions/fetchAction'
import SignUpForm from '../../components/SignupForm/SignupForm'

class SignUp extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            username: '',
            email: '',
            password: '',
            location: '',
            role: 'user'
        };
        this.onChange = this.onChange.bind(this)
        this.onSubmit= this.onSubmit.bind(this)
    }
onChange(e){
this.setState({[e.target.name]: e.target.value})
}
onSubmit(e){
    e.preventDefault();
    const user = {
        username: this.state.username,
        email: this.state.email,
        password: this.state.password,
        location: this.state.location,
        role: this.state.role
    };
    this.props.registerAction(user)
}
    render() {
        return (
            <SignUpForm />
        );
    }
}

export default connect (
    null,
    {registerAction}
) (SignUp);
