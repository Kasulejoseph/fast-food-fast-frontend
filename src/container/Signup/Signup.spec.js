import React from "react";
import { shallow } from "enzyme";
import {SignUp} from "./Signup";
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

let props = {
  onChange: jest.fn(),
  onSubmit: jest.fn(),
};
describe("Test SignUp", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<SignUp {...props} />);
  });
  it("LogIn snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("test handle onsubmit event", () => {      
    wrapper.instance().onSubmit({ preventDefault() {} });
    expect(wrapper.state()).toEqual({"email": "", "location": "", "password": "", "role": "user", "username": ""});
  });

  it("should handle onchange function", () => {
    wrapper
      .instance()
      .onChange({ target: { name: "email", value: "kasule@andela.com" } });
    expect(wrapper.state("email")).toEqual("kasule@andela.com");
  });

});