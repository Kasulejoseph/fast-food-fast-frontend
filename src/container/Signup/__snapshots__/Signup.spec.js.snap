// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SignUp LogIn snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUp
    onChange={[MockFunction]}
    onSubmit={[MockFunction]}
    registerAction={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <NavBarAuth
          active2="de-active"
          isOpen={false}
          onClick={[Function]}
        />,
        ",",
        <SignupForm
          onChange={[Function]}
          onSubmit={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "active2": "de-active",
          "isOpen": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      ",",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onChange": [Function],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <NavBarAuth
            active2="de-active"
            isOpen={false}
            onClick={[Function]}
          />,
          ",",
          <SignupForm
            onChange={[Function]}
            onSubmit={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "active2": "de-active",
            "isOpen": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        ",",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onChange": [Function],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
